### PR TITLE
Cleanup lowpass.c // replace by pt1

### PR DESCRIPTION
**Pull request by borisbstyle**
This will cleanup some code. I think pt1 is enough for vbat and servo

@ledvinap 
What do you think?

### CODE REVIEW

**Comment by martinbudden** [comment]
Looks like an undefined reference problem. You need to include the stubs:
```
uint32_t micros(void) {return 0;}
float filterApplyPt1(float input, filterStatePt1_t *filter, uint8_t f_cut, float dT) {UNUSED(input);UNUSED(filter);UNUSED(f_cut);UNUSED(dT);return 0.0f;}
```

in the stubs section at the bottom of `battery_unittest.cc`. Additionally you'll need to add

```#include "common/filter.h"```

in the `extern "C"` section at the top.

**Comment by borisbstyle** [comment]
hmmm....still failing. That linking doesnt seem right.

**Comment by martinbudden** [comment]
Yes, it looks like you have included `common/filter.h` twice, in the extern section at the top and an the one at the bottom of the test code. It should only be included at the top. It also seems that  `common/filter.h` does not include a `#pragma once` directive.

**Comment by martinbudden** [comment]
OK, it's now compiling, but failing the `BatteryTest.BatteryState` tests. Looks like the simple stub I gave you is not sufficient.

```
float filterApplyPt1(float input, filterStatePt1_t *filter, uint8_t f_cut, float dT) {UNUSED(filter);UNUSED(f_cut);UNUSED(dT);return input;}
```
Might work, but again I need to get the code onto my machine to look at it in detail.

It's getting late, so I'll have a look tomorrow.

**Comment by borisbstyle** [comment]
thanks.

can you explain me more how to run the tests locally?
What exactly do I need. travis scripts do run tests, but those always fail
anyway on my machine.

I would be able to also write some tests for these features.
Op 11 feb. 2016 00:52 schreef "Martin Budden" <notifications@github.com>:

**Comment by martinbudden** [comment]
Getting test you run on your machine is relatively straightforward if you can already build using `make`. Just change directory into the test directory (ie `cd src/test`) and then run `make` from there. This will build the tests. Use `make test` to run the tests.

**Comment by borisbstyle** [comment]
@martinbudden 
Ah....stupid me. I didnt realise the tests have their own build environment! It runs fine. Now I can debug

**Comment by ledvinap** [comment]
@martinbudden
`make test` in topmost directory is all that is needed ...
`make all_targets` is useful to test that all targets build correctly

**Comment by ledvinap** [comment]
@borisbstyle : It would be great to unified filtering interface and use in consistently in code.

Removing second order LPF will save quite a lot of flash.

First order LPF should be enough for battery (1Hz cutoff may change voltage compensation behavior, but that is not merged yet).

Change of servo filtering should be tested.
Biquad filter may be used too, it is almost identical to removed lowpass filter.

**Comment by borisbstyle** [comment]
@ledvinap 
I tested the battery and it works great! I compared the data with old lowpass 10hz and 1hz IIR and not much difference through batteries. For battery this is really enough. I might perhaps add interpolation of vbat voltage simillar to rc smoothing as now it shows stepping.
I also tested it with the other PR

I obviously cannot test servo filtering, but I could change that one to biquad. Was thinking about that, but wanted to hear feedback from someone who has more experience with servos.

**Comment by ledvinap** [comment]
Old lowpass was 2nd order IIR (butterworth), PT1 is first order.

Is there any advantage for vbat interpolation? IMO it is only cosmetic effect.
If voltage is part of control loop and steps are causing proglems, it would be better/easier to filter at loop frequency.

**Comment by borisbstyle** [comment]
@ledvinap 
I didn't notice any negative effect of stepping at all. It totally seems cosmetic. 1st order is more than fine for the battery. It is also easier to implement as battery is not being constantly updated due to scheduler. Not that it makes that much difference, but its cleaner IMO. 

For servo it totally makes sense to add biquad

File: src/main/flight/mixer.c [hunk 1]
```diff
@@ -979,6 +980,8 @@ void filterServos(void)
 {
 #ifdef USE_SERVOS
     int16_t servoIdx;
+    static bool servoFilterIsSet;
+    biquad_t servoFitlerState[MAX_SUPPORTED_SERVOS];
```
Comment (hydra) [inline src/main/flight/mixer.c]: typo: 'servoFilterState'

File: src/main/flight/mixer.c [hunk 2]
```diff
@@ -986,7 +989,11 @@ void filterServos(void)
 
     if (mixerConfig->servo_lowpass_enable) {
         for (servoIdx = 0; servoIdx < MAX_SUPPORTED_SERVOS; servoIdx++) {
-            servo[servoIdx] = (int16_t)lowpassFixed(&lowpassFilters[servoIdx], servo[servoIdx], mixerConfig->servo_lowpass_freq);
+            if (!servoFilterIsSet) {
```
Comment (hydra) [inline src/main/flight/mixer.c]: this code only works for the first servo.  the state is never initialised for servoIdx > 0.

File: src/main/flight/mixer.c [hunk 3]
```diff
@@ -60,6 +61,7 @@ uint8_t motorCount;
 
 int16_t motor[MAX_SUPPORTED_MOTORS];
 int16_t motor_disarmed[MAX_SUPPORTED_MOTORS];
+extern float dT;
```
Comment (hydra) [inline src/main/flight/mixer.c]: not needed it seems.

**Pull request merged**